### PR TITLE
Adds consensus tests

### PR DESCRIPTION
**Pull request by InfiniteSwerve**
# Problem 
Currently we have no way to guarantee our consensus works properly
# Solution
Add consensus tests via the interface provided by `consensus.mli`. Note these tests are **not** comprehensive, but should be good enough until we have generative testing.

### CODE REVIEW

File: src/consensus/tests/test_consensus.ml [hunk 1]
```diff
@@ -0,0 +1,717 @@
...
+  List.fold_left
+    (fun pool (block, votes) ->
+      let (Block { level; _ }) = block in
+      let pool = append_block ~block pool in
+      List.fold_left (fun pool vote -> append_vote ~level ~vote pool) pool votes)
+    pool blocks
+
+let time second = Timestamp.of_float second
+let next_timeout until = time (until +. Deku_constants.block_timeout)
+let ensure msg actual = Alcotest.(check' bool) ~msg ~expected:true ~actual
```
[the remark this answers is not in the document]
Comment (InfiniteSwerve) [reply src/consensus/tests/test_consensus.ml:77]: Sorry for the late reply, I had thought this was merged based on our in-person discussion. I'll rebase to main and update the tests to account for the latest consensus API, then, if there's no more issues, please approve it.

**Comment by InfiniteSwerve** [comment]
@EduardoRFS Branch has been rebased

**Comment by EduardoRFS** [comment]
anime girl giving a start to Jack

File: src/consensus/tests/test_consensus.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (EduardoRFS) [reply src/consensus/tests/test_consensus.ml]: Please open an issue about this

[the remark this answers is not in the document]
Comment (InfiniteSwerve) [reply src/consensus/tests/test_consensus.ml]: [Done](https://github.com/marigold-dev/deku/issues/956)

**Pull request merged**